src/components/TimeGrid.jsx

```diff
@@ -0,0 +1,238 @@
+import React, { PropTypes, Component } from 'react'
+import cn from 'classnames'
+import localizer from '../localizer.js'
+import formats from '../formats.js'
+import { findDOMNode } from 'react-dom';
+import classes from 'dom-helpers/class';
+import scrollbarSize from 'dom-helpers/util/scrollbarSize';
+import getWidth from 'dom-helpers/query/width';
+
+import TimeGridHeader from './TimeGridHeader.jsx'
+import TimeGutter from './TimeGutter.jsx'
+import DaySlot from './DaySlot.jsx'
+import TimeGridAllDay from './TimeGridAllDay.jsx'
+
+import EventRow from '../EventRow.jsx'
+import EventCell from '../EventCell.jsx'
+
+import { accessor as get } from '../utils/accessors';
+import dates from '../utils/dates';
+import {
+  inRange, eventSegments, endOfRange
+  , eventLevels, sortEvents, segStyle } from '../utils/eventLevels';
+import { elementType, accessor } from '../utils/propTypes';
+
+const MIN_ROWS = 2;
+
+export default class TimeGrid extends Component {
+  static propTypes = {
+    ...DaySlot.propTypes,
+    ...TimeGutter.propTypes,
+
+    end: PropTypes.instanceOf(Date),
+    start: PropTypes.instanceOf(Date),
+    selectRangeFormat: PropTypes.func.isRequired,
+    eventTimeRangeFormat: PropTypes.func.isRequired,
+    timeGutterFormat: PropTypes.string,
+    dayFormat: PropTypes.string.isRequired,
+    culture: PropTypes.string.isRequired,
+    components: PropTypes.shape({
+      event: elementType,
+
+      toolbar: PropTypes.element,
+
+      agenda: PropTypes.shape({
+        date: elementType,
+        time: elementType,
+        event: elementType
+      }),
+
+      day: PropTypes.shape({ event: elementType }),
+      week: PropTypes.shape({ event: elementType }),
+      month: PropTypes.shape({ event: elementType })
+    }),
+
+    allDayAccessor: accessor,
+    titleAccessor: accessor,
+    startAccessor: accessor,
+    endAccessor: accessor
+  }
+
+  static defaultProps = {
+    ...DaySlot.defaultProps,
+    ...TimeGutter.defaultProps,
+    culture: 'en',
+    selectRangeFormat: formats().selectRangeFormat,
+    eventTimeRangeFormat: formats().eventTimeRangeFormat,
+    components: {
+      event: null
+    },
+    allDayAccessor: 'allDay',
+    startAccessor: 'start',
+    endAccessor: 'end',
+    titleAccessor: 'title'
+  }
+
+  _adjustGutter() {
+    const isRtl = this.props.rtl;
+    const header = this._headerCell;
+    const gutterCells = [findDOMNode(this.refs.gutter), ...this._gutters]
+    const isOverflowing = this.refs.content.scrollHeight > this.refs.content.clientHeight;
+
+    let width = this._gutterWidth
```

Comment:
makes sense.  I left this code unchanged to minimize the changes to process.  Plus it already works.  What we really need is the new grid css, but oh well :)